recce_cloud/cli.py

```diff
@@ -25,5 +32,213 @@ def version():
+    - Each session stores manifests/catalogs from both environments
+    - Sessions are linked to PRs/MRs for team collaboration and review
+    - This enables automated PR gating and impact analysis in CI/CD
+
+    \b
+    Authentication:
+    Uses environment variables for authentication:
+    - RECCE_API_TOKEN: Recce Cloud API token (primary)
+    - GITHUB_TOKEN: GitHub token (fallback, commonly available in GitHub Actions)
+
+    \b
+    Auto-Detection:
+    This command automatically detects your environment:
+    - SCM Provider: GitHub, GitLab, Bitbucket, or generic git
+    - CI/CD Platform: GitHub Actions, GitLab CI, Bitbucket Pipelines, CircleCI, etc.
+    - Git Information: repository, branch, commit hash
+    - PR/MR Information: ID, title, URL (used to link session to PR/MR)
+
+    \b
+    Environment Variables:
+    - RECCE_SESSION_ID: Target session ID for upload (required)
+    - RECCE_API_TOKEN: Recce Cloud API token
+    - GITHUB_TOKEN: GitHub token (fallback for authentication)
+    - RECCE_CLOUD_API_HOST: Custom API host (default: https://cloud.datarecce.io)
+
+    \b
+    Examples:
+    # Upload from default target directory with session ID from env
+    export RECCE_SESSION_ID=abc123
+    recce-cloud upload
+
+    \b
+    # Upload from custom target path
+    recce-cloud upload --target-path my-target --session-id abc123
+
+    \b
+    # Typical CI/CD usage (GitHub Actions)
+    recce-cloud upload --session-id ${{ steps.create-session.outputs.session-id }}
+
+    \b
+    Exit Codes:
+    0 - Success
+    1 - Environment detection error
+    2 - Authentication error
+    3 - File validation error (missing or invalid manifest/catalog)
+    4 - Upload error
+    """
+    console = Console()
+
+    # Validate session ID
+    if not session_id:
+        console.print("[red]Error:[/red] Session ID is required")
+        console.print("Provide --session-id or set RECCE_SESSION_ID environment variable")
+        sys.exit(2)
+
+    # 1. Validate artifacts exist
+    if not verify_artifacts_path(target_path):
+        console.print(f"[red]Error:[/red] Invalid target path: {target_path}")
+        console.print("Please provide a valid target path containing manifest.json and catalog.json.")
+        sys.exit(3)
+
+    manifest_path = os.path.join(target_path, "manifest.json")
+    catalog_path = os.path.join(target_path, "catalog.json")
+
+    # 2. Extract adapter type from manifest
+    try:
+        adapter_type = get_adapter_type(manifest_path)
+    except Exception as e:
+        console.print("[red]Error:[/red] Failed to parse adapter type from manifest.json")
+        console.print(f"Reason: {e}")
+        sys.exit(3)
+
+    # 3. Get authentication token
+    token = os.getenv("RECCE_API_TOKEN") or os.getenv("GITHUB_TOKEN")
+    if not token:
+        console.print("[red]Error:[/red] No authentication token provided")
+        console.print("Set RECCE_API_TOKEN or GITHUB_TOKEN environment variable")
```

Comment:
I think we can just mention `RECCE_API_TOKEN` in the console message.

Reply:
good point, I removed it and removed tests as well.